<launch>
    <arg name="rviz" default="true"/>

    <arg name="calibration_file" default="$(find imu_gnss_fusion)/config/camchain-imucam-euroc.yaml" />    

    <!-- <param name="acc_noise"       type="double" value="0.05" />
    <param name="gyr_noise"       type="double" value="0.005" />
    <param name="acc_bias_noise"  type="double" value="0.01" />
    <param name="gyr_bias_noise"  type="double" value="0.001" /> -->

    <param name="acc_noise"       type="double" value="0.08" />
    <param name="gyr_noise"       type="double" value="0.004" />
    <param name="acc_bias_noise"  type="double" value="0.00004" />
    <param name="gyr_bias_noise"  type="double" value="2.0e-6" />

    <param name="topic_vo" value="/odom_orbslam" />
    <param name="topic_imu" value="/imu0" />
    
    <node name="imu_vo_fusion_node" pkg="imu_gnss_fusion" type="imu_vo_fusion_node" output="screen">
      <rosparam command="load" file="$(arg calibration_file)"/>
    </node>

    <group if="$(arg rviz)">
      <node pkg="rviz" type="rviz" name="rviz" output="screen" 
        args="-d $(find imu_gnss_fusion)/rviz/imu_vo_fusion.rviz" required="true">
      </node>
    </group>
</launch>